Source/Core/DolphinQt/Settings/InterfacePane.cpp

```diff
@@ -169,23 +170,36 @@ void InterfacePane::CreateInGame()
   m_checkbox_enable_osd = new QCheckBox(tr("Show On-Screen Display Messages"));
   m_checkbox_show_active_title = new QCheckBox(tr("Show Active Title in Window Title"));
   m_checkbox_pause_on_focus_lost = new QCheckBox(tr("Pause on Focus Loss"));
-  m_checkbox_hide_mouse = new QCheckBox(tr("Always Hide Mouse Cursor"));
-  m_checkbox_lock_mouse = new QCheckBox(tr("Lock Mouse Cursor"));
 
-  m_checkbox_hide_mouse->setToolTip(
-      tr("Will immediately hide the Mouse Cursor when it hovers on top of the Render Widget, "
-         "otherwise "
-         "there is a delay.\nIf \"Lock Mouse Cursor\" is enabled, it will hide on Mouse locked"));
+  auto* mouse_groupbox = new QGroupBox(tr("Mouse Cursor Visibility"));
+  auto* m_vboxlayout_hide_mouse = new QVBoxLayout;
+  mouse_groupbox->setLayout(m_vboxlayout_hide_mouse);
+
+  m_radio_cursor_visible_movement = new QRadioButton(tr("On Movement"));
+  m_radio_cursor_visible_movement->setToolTip(
+      tr("Mouse Cursor hides after inactivity and returns upon Mouse Cursor movement."));
```

Comment:
In the "On Movement" and "Never" mode, is it like before where the cursor became invisible only when on top of the render widget and always staeyd visible outside of it (e.g. if playing in a window or with multiple monitors)?
Or has that changed? If so, shouldn't there be yet another setting "only hide when on top of render widget"?
Please make sure this works well with and without mouse lockign?

Reply:
Correct, the cursor only becomes invisible when on top of the render widget and when it is inactive. I don't think an additional settings should be added to change that behavior. It doesn't seem significant.

I did test mouse locking and it is working as intended with my changes.